fix: DA client connection flow

EmandM opened the pull request:
<!-- Replace this block with what this PR does and why. Describe what you'd like reviewers to know, how you applied the engineering principles, and any interesting tradeoffs made. Delete bullet points below that don't apply, and update the changelog section as appropriate. -->

<!-- Add short version of the JIRA ticket to the PR title (e.g. "feat: new shiny feature [MTT-123]") -->
[MTT-12308](https://jira.unity3d.com/browse/MTT-12308)

## Changelog

- Fixed: Initial connection flow with CMB Service and scene management enabled

## Testing and Documentation

- Includes unit tests.

<!--  Uncomment and mark items off with a * if this PR deprecates any API:
### Deprecated API
- [ ] An `[Obsolete]` attribute was added along with a `(RemovedAfter yyyy-mm-dd)` entry.
- [ ] An [api updater] was added.
- [ ] Deprecation of the API is explained in the CHANGELOG.
- [ ] The users can understand why this API was removed and what they should use instead.
-->

## Backport

<!-- If this is a backport:
 - Add the following to the PR title: "\[Backport\] ..." .
 - Link to the original PR.
If this needs a backport - state this here
If a backport is not needed please provide the reason why.
If the "Backports" section is not present it will lead to a CI test failure. 
-->

Not needed.

Fixes an issue with CMB Service so this is 2.0 only functionality.

EmandM committed:
fix: DA client connection flow

EmandM committed:
Fix whitespace issues

NoelStephensUnity commented on com.unity.netcode.gameobjects/Runtime/SceneManagement/NetworkSceneManager.cs:
```diff
@@ -2603,14 +2603,6 @@ private void HandleSessionOwnerEvent(uint sceneEventId, ulong clientId)
                         // Mark this client as being connected
                         NetworkManager.ConnectedClients[clientId].IsConnected = true;
 
-                        // Notify the local server that a client has finished synchronizing
```
I think this needs to stay here?
Only clients in a distributed authority topology will invoke the HandleSessionOwnerEvent, but they also need to be notified of all scene events in order to stay synchronized with the current session.

Let's walk through this so I can better understand your thoughts/direction on this?

NoelStephensUnity reviewed (commented):
Just need a walk through on the synchronization event.

unity-renovate[bot] committed:
chore(deps): update dependency recipeengine.modules.wrench to 0.10.50 (develop-2.0.0) (#3461)

This PR contains the following updates:

| Package | Type | Update | Change |
|---|---|---|---|
| RecipeEngine.Modules.Wrench | nuget | patch | `0.10.49` -> `0.10.50` |

---

### Configuration

📅 **Schedule**: Branch creation - At any time (no schedule defined),
Automerge - At any time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you
are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about this update
again.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR has been generated by [Unity Renovate
Bot](https://internaldocs.unity.com/renovate/).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIwLjAuMC1zZW1hbnRpYy1yZWxlYXNlIiwidXBkYXRlZEluVmVyIjoiMC4wLjAtc2VtYW50aWMtcmVsZWFzZSIsInRhcmdldEJyYW5jaCI6ImRldmVsb3AtMi4wLjAiLCJsYWJlbHMiOltdfQ==-->

## Backport

https://github.com/Unity-Technologies/com.unity.netcode.gameobjects/pull/3460#pullrequestreview-2874656403

Co-authored-by: unity-renovate[bot] <120015202+unity-renovate[bot]@users.noreply.github.com>

unity-renovate[bot] committed:
chore(deps): update dependency recipeengine.modules.wrench to 0.11.1 (develop-2.0.0) (#3464)

This PR contains the following updates:

| Package | Type | Update | Change |
|---|---|---|---|
| RecipeEngine.Modules.Wrench | nuget | minor | `0.10.50` -> `0.11.1` |

---

### Configuration

📅 **Schedule**: Branch creation - At any time (no schedule defined),
Automerge - At any time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you
are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about this update
again.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR has been generated by [Unity Renovate
Bot](https://internaldocs.unity.com/renovate/).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIwLjAuMC1zZW1hbnRpYy1yZWxlYXNlIiwidXBkYXRlZEluVmVyIjoiMC4wLjAtc2VtYW50aWMtcmVsZWFzZSIsInRhcmdldEJyYW5jaCI6ImRldmVsb3AtMi4wLjAiLCJsYWJlbHMiOltdfQ==-->

## Backport

https://github.com/Unity-Technologies/com.unity.netcode.gameobjects/pull/3463

Co-authored-by: unity-renovate[bot] <120015202+unity-renovate[bot]@users.noreply.github.com>

EmandM committed:
Move expected message to ConnectionApproved

EmandM committed:
Fix compilation errors

EmandM committed:
Fix OnVerifyCanReceive

EmandM committed:
only send SynchronizeComplete to the rust

EmandM committed:
remove duplicate peer connected callback

EmandM committed:
Fix whitespace error

EmandM committed:
Merge branch 'develop-2.0.0' of https://github.com/Unity-Technologies/com.unity.netcode.gameobjects into fix/mttb-12308/client-connection-flow

EmandM committed:
Remove unnecessary changes

EmandM committed:
Fix tests

EmandM committed:
Add actual feature flag for this feature rather than piggybacking of the existing feature flag

EmandM committed:
Remove test changes

EmandM committed:
Merge branch 'develop-2.0.0' of https://github.com/Unity-Technologies/com.unity.netcode.gameobjects into fix/mttb-12308/client-connection-flow

NoelStephensUnity committed:
Merge branch 'develop-2.0.0' into fix/mttb-12308/client-connection-flow

EmandM committed:
Fix Connection deferred queue

EmandM committed:
Log warning on duplicate connection approved

NoelStephensUnity commented on com.unity.netcode.gameobjects/Runtime/SceneManagement/NetworkSceneManager.cs:
```diff
@@ -2554,27 +2533,29 @@ private void HandleSessionOwnerEvent(uint sceneEventId, ulong clientId)
-                                    break;
-                                }
-                            }
+                            // Debug.Log($"Remove client-{clientId} from synchronization queue.");
```
Thoughts on removing commented out code?

NoelStephensUnity reviewed (approved):
Nice integration tests... :godmode: 

- The PR still needs a "Fix" change log entry.
- There is a question about the commented out code block.

Other than those two minor things...
Looks good to me!
👍

EmandM committed:
Remove commented out code and add entry to changelog

NoelStephensUnity committed:
Merge branch 'develop-2.0.0' into fix/mttb-12308/client-connection-flow

netcode-ci-service committed:
Merge develop-2.0.0 into fix/mttb-12308/client-connection-flow

netcode-ci-service committed:
Merge develop-2.0.0 into fix/mttb-12308/client-connection-flow
